src/apatheia/jobs.nim

```diff
@@ -0,0 +1,82 @@
+import std/tables
+import std/macros
+import std/sequtils
+
+import ./queues
+
+import taskpools
+import chronos
+
+export queues
+
+## This module provides a simple way to submit jobs to taskpools
+## and getting a result returned via an async future.
+## 
+##
+
+type
+  JobId* = uint ## job id, should match `future.id()`
+
+  JobQueue*[T] = ref object
+    ## job queue object
+    queue*: SignalQueue[(JobId, T)]
+    futures*: Table[JobId, Future[T]]
+    taskpool*: Taskpool
+    running*: bool
+
+  JobResult*[T] = object
+    ## hold a job result to be returned by jobs
+    id*: JobId
+    queue*: SignalQueue[(JobId, T)]
+
+proc processJobs*[T](jobs: JobQueue[T]) {.async.} =
+  ## Starts a "detached" async processor for a given job queue.
+  ## 
+  ## This processor waits for events from the queue in the JobQueue
+  ## and complete the associated futures.
+
+  while jobs.running:
+    echo "jobs running..."
+    let res = await(jobs.queue.wait()).get()
+    echo "jobs result: ", res.repr
+    echo "jobs futes: ", jobs.futures.unsafeAddr.pointer.repr, " => ", jobs.futures.keys().toSeq()
```

Comment:
Maybe add a chronicles dependency and use trace/debug for this?

Reply:
@tbekas that's a good idea. Some of this is just initial debugging and to be removed, but we'll definitely want to log certain events.